### ISSUE TITLE
“Bad request” error message is never shown

### ISSUE THREAD

**Issue by atwixfirster** [magento/magento2#24330]
### Summary (*)

Customer confirmation process. “Bad request” message is never shown.

### Examples (*)

**Preconditions**
1. Set `Require Emails Confirmation` option to `Yes` on back-end side

<img width="1536" alt="back-end" src="https://user-images.githubusercontent.com/13585327/63840660-22524e80-c98a-11e9-8ef0-7791c07cf505.png">

**Steps to reproduce**
1. Create new customer

2. You will get a confirmation link in the email

3. Set empty customer id parameter in the confirmation URL.

For example, confirmation URL is

```
http://<MAGENTO-INSTANCE>/customer/account/confirm/?id=XXX&key=3D384de80ffb30003a72d29e5326932547
```

Remove XXX from the provided URL:

```
http://<MAGENTO-INSTANCE>/customer/account/confirm/?id=&key=3D384de80ffb30003a72d29e5326932547
```

4. Open URL in browser

```
http://<MAGENTO-INSTANCE>/customer/account/confirm/?id=&key=3D384de80ffb30003a72d29e5326932547
```

**Expected result:** `Bad request` message is displayed for customer

<img width="634" alt="expected" src="https://user-images.githubusercontent.com/13585327/63840534-e6b78480-c989-11e9-8f92-821a8f36ee35.png">

**Actual result:** `There was an error confirming the account` message is displayed for customer

<img width="627" alt="actual" src="https://user-images.githubusercontent.com/13585327/63840547-ecad6580-c989-11e9-9c1c-efd4be0520f0.png">

### Proposed solution

https://github.com/magento/magento2/pull/24331

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @dmytro-ch 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19695` were created

**Issue Available**: @dmytro-ch, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by dmytro-ch** [comment]
Hi @atwixfirster,
this is rather an improvement request than bug report.
Feel free to create the pull requests with minor fixes and improvements without opening the issues.

Thank you!

**Comment by magento-engcom-team** [comment]
Hi @atwixfirster. Thank you for your report. 
The issue has been fixed in magento/magento2#24331 by @atwixfirster in 2.3-develop branch
Related commit(s):
 - [40c0e2859b1bf4f5795a403861dd231b1e2483ae](https://github.com/magento/magento2/commit/40c0e2859b1bf4f5795a403861dd231b1e2483ae)
 - [e5869014a8e7ea98cf3a6bb5a822c2683d2841a2](https://github.com/magento/magento2/commit/e5869014a8e7ea98cf3a6bb5a822c2683d2841a2)

The fix will be available with the upcoming 2.3.4 release.
